ansible/roles/test/files/helpers/ferret.py

```diff
@@ -170,16 +170,29 @@ def action(self, interface):
         ext_eth_type = data[0x0c:0x0e]
         if ext_eth_type != binascii.unhexlify('0800'):
             print "Not 0x800 eth type"
+            self.hexdump(data)
+            print
             return
         src_ip = data[0x001a:0x001e]
         dst_ip = data[0x1e:0x22]
         gre_flags = data[0x22:0x24]
         gre_type  = data[0x24:0x26]
-        # FIXME: check gre type and gre_flags
 
-        arp_request = data[0x26:]
+        gre_type_r = struct.unpack('!H', gre_type)[0]
+        if gre_type_r == 0x88be:   # Broadcom
+            arp_request = data[0x26:]
```

Comment:
>0x26 [](start = 31, length = 4)

magic number, how about giving a name?

Reply:
There are too many "magic" numbers. The name of the variables are the names for the numbers.